controllers/tasks.js

```diff
@@ -334,38 +322,40 @@ const updateTaskStatus = async (req, res, next) => {
     let userStatusUpdate;
     const taskId = req.params.id;
     const { userStatusFlag } = req.query;
+    const { status } = req.body;
     const { id: userId, username } = req.userData;
     const task = await tasks.fetchSelfTask(taskId, userId);
 
     if (task.taskNotFound) return res.boom.notFound("Task doesn't exist");
     if (task.notAssignedToYou) return res.boom.forbidden("This task is not assigned to you");
-    if (task.taskData.status === TASK_STATUS.VERIFIED || req.body.status === TASK_STATUS.MERGED)
+    if (
+      task.taskData.status === TASK_STATUS.VERIFIED ||
+      status === TASK_STATUS.MERGED ||
+      status === TASK_STATUS.BACKLOG
+    )
       return res.boom.forbidden("Status cannot be updated. Please contact admin.");
 
     if (userStatusFlag) {
       if (task.taskData.status === TASK_STATUS.DONE && req.body.percentCompleted < 100) {
-        if (req.body.status === TASK_STATUS.DONE || !req.body.status) {
+        if (status === TASK_STATUS.DONE || !status) {
           return res.boom.badRequest("Task percentCompleted can't updated as status is DONE");
         }
       }
 
-      if (
-        (req.body.status === TASK_STATUS.DONE || req.body.status === TASK_STATUS.VERIFIED) &&
-        task.taskData.percentCompleted !== 100
-      ) {
+      if ((status === TASK_STATUS.DONE || status === TASK_STATUS.VERIFIED) && task.taskData.percentCompleted !== 100) {
         if (req.body.percentCompleted !== 100) {
           return res.boom.badRequest("Status cannot be updated. Task is not done yet");
         }
       }
     }
 
     if (task.taskData.status === TASK_STATUS.COMPLETED && req.body.percentCompleted < 100) {
-      if (req.body.status === TASK_STATUS.COMPLETED || !req.body.status) {
+      if (status === TASK_STATUS.COMPLETED || !status) {
```

Comment:
!status should be checked first so that JS program short circuits from that point onwards and it has not run the other condition

Reply:
done